bundles/org.openhab.binding.metofficedatahub/src/main/java/org/openhab/binding/metofficedatahub/internal/MetOfficeDataHubSiteApiHandler.java

```diff
@@ -0,0 +1,839 @@
+
+            updateState(channelPrefix + SITE_DAILY_DAY_MAX_SCREEN_TEMPERATURE,
+                    getQuantityTypeState(data.getDayMaxScreenTemperature(), SIUnits.CELSIUS));
+
+            updateState(channelPrefix + SITE_DAILY_NIGHT_MIN_SCREEN_TEMPERATURE,
+                    getQuantityTypeState(data.getNightMinScreenTemperature(), SIUnits.CELSIUS));
+
+            updateState(channelPrefix + SITE_DAILY_DAY_PROBABILITY_OF_PRECIPITATION,
+                    getQuantityTypeState(data.getDayProbabilityOfPrecipitation(), Units.PERCENT));
+
+            updateState(channelPrefix + SITE_DAILY_NIGHT_PROBABILITY_OF_PRECIPITATION,
+                    getQuantityTypeState(data.getNightProbabilityOfPrecipitation(), Units.PERCENT));
+
+            updateState(channelPrefix + SITE_DAILY_DAY_PROBABILITY_OF_SNOW,
+                    getQuantityTypeState(data.getDayProbabilityOfSnow(), Units.PERCENT));
+
+            updateState(channelPrefix + SITE_DAILY_NIGHT_PROBABILITY_OF_SNOW,
+                    getQuantityTypeState(data.getNightProbabilityOfSnow(), Units.PERCENT));
+
+            updateState(channelPrefix + SITE_DAILY_DAY_PROBABILITY_OF_HEAVY_SNOW,
+                    getQuantityTypeState(data.getDayProbabilityOfHeavySnow(), Units.PERCENT));
+
+            updateState(channelPrefix + SITE_DAILY_NIGHT_PROBABILITY_OF_HEAVY_SNOW,
+                    getQuantityTypeState(data.getNightProbabilityOfHeavySnow(), Units.PERCENT));
+
+            updateState(channelPrefix + SITE_DAILY_DAY_PROBABILITY_OF_RAIN,
+                    getQuantityTypeState(data.getDayProbabilityOfRain(), Units.PERCENT));
+
+            updateState(channelPrefix + SITE_DAILY_NIGHT_PROBABILITY_OF_RAIN,
+                    getQuantityTypeState(data.getNightProbabilityOfRain(), Units.PERCENT));
+
+            updateState(channelPrefix + SITE_DAILY_DAY_PROBABILITY_OF_HEAVY_RAIN,
+                    getQuantityTypeState(data.getDayProbabilityOfHeavyRain(), Units.PERCENT));
+
+            updateState(channelPrefix + SITE_DAILY_NIGHT_PROBABILITY_OF_HEAVY_RAIN,
+                    getQuantityTypeState(data.getNightProbabilityOfHeavyRain(), Units.PERCENT));
+
+            updateState(channelPrefix + SITE_DAILY_DAY_PROBABILITY_OF_HAIL,
+                    getQuantityTypeState(data.getDayProbabilityOfHail(), Units.PERCENT));
+
+            updateState(channelPrefix + SITE_DAILY_NIGHT_PROBABILITY_OF_HAIL,
+                    getQuantityTypeState(data.getNightProbabilityOfHail(), Units.PERCENT));
+
+            updateState(channelPrefix + SITE_DAILY_DAY_PROBABILITY_OF_SFERICS,
+                    getQuantityTypeState(data.getDayProbabilityOfSferics(), Units.PERCENT));
+
+            updateState(channelPrefix + SITE_DAILY_NIGHT_PROBABILITY_OF_SFERICS,
+                    getQuantityTypeState(data.getNightProbabilityOfSferics(), Units.PERCENT));
+        }
+    }
+
+    private void handleNon200Response(final Response resp) {
```

Comment:
This is the responsibility of the bridge. As the bridge is using the apikey. The `site` is only a client/user of this. For exmaple, if the bridge has a invalid apikey, _all_ child things need to go offline, not just this one.